### ISSUE TITLE
Cheaper forum transactions through category depth witness

### ISSUE THREAD

**Issue by bedeho** [Joystream/joystream#2541]
Right now, the weight function for the `Olympia` runtime charges the caller as if the object in question is at the maximum category depth, however this is a bit uncessaray, and can add a bit of cost if we want to allow uncapped category depth, or even remove the concept entirely. The remedy wuold be to add category depth witness to the extrinsics, ued in the benchmarking, and verified in the extrinsic.

┆Issue is synchronized with this [Asana task](https://app.asana.com/0/1201703952879841/1201704745789611) by [Unito](https://www.unito.io)

**Comment by iorveth** [comment]
Good point, this can be calculated and therefore used to tighten weights.

**Comment by bedeho** [comment]
Perhaps we can just remove the category depth limit entirely?
